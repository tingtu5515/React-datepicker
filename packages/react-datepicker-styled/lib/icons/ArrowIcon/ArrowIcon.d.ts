/// <reference types="./src/@types" />
interface Props extends IconProps {
  direction?: 'up' | 'down' | 'left' | 'right'
}
declare function ArrowIcon({height, width, color, direction, className}: Props): JSX.Element
export default ArrowIcon
