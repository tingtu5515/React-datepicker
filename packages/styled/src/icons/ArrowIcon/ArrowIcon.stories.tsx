import React from 'react'
import {storiesOf} from '@storybook/react'
import {text, color, select} from '@storybook/addon-knobs'
import ArrowIcon from '.'

const IconDirections = {
  left: 'left',
  right: 'right',
  up: 'up',
  down: 'down',
}

storiesOf('Components/Icons', module).add('ArrowIcon', () => (
  <ArrowIcon
    iconColor={color('Color', '#000')}
    height={text('Height', '30px')}
    width={text('Width', '30px')}
    // @ts-ignore
    direction={select('Direction', IconDirections, 'right')}
  />
))
