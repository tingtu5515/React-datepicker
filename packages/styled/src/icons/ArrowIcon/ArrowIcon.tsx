import React from 'react'

interface Props {
  direction?: 'up' | 'down' | 'left' | 'right'
  height: string
  width: string
  iconColor?: string
}

function calculateAngle(direction: Props['direction']) {
  switch (direction) {
    case 'up':
      return 0
    case 'down':
      return 180
    case 'left':
      return -90
    case 'right':
    default:
      return 90
  }
}

function ArrowIcon({height, width, iconColor, direction = 'right'}: Props) {
  const angle = calculateAngle(direction)
  return (
    <svg
      width={width}
      height={height}
      color={iconColor}
      transform={`rotate(${angle} 0 0)`}
      viewBox="0 0 9 12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fill="currentColor"
        d="M4.46.001a.538.538 0 0 0-.358.174L.156 4.48a.538.538 0 1 0 .796.724l3.01-3.285v13.689a.563.563 0 0 0 .538.55.563.563 0 0 0 .538-.55V1.918l3.01 3.286a.538.538 0 1 0 .796-.724L4.898.175a.538.538 0 0 0-.437-.174z"
      />
    </svg>
  )
}

export default ArrowIcon
