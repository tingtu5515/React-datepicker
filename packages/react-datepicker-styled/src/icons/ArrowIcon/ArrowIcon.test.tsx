import React from 'react'
import {render} from 'react-testing-library'
import ArrowIcon from '.'

test('matches snapshot', () => {
  const {container} = render(<ArrowIcon height="14px" width="14px" color="red" />)
  expect(container).toMatchSnapshot()
})

test('supports direction', () => {
  const {container} = render(<ArrowIcon direction="left" height="14px" width="14px" color="red" />)
  expect(container).toMatchSnapshot()
})
