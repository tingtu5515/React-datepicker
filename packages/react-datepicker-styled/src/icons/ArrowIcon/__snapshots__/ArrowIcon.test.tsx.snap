// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <svg
    class=""
    color="red"
    height="14px"
    transform="rotate(90 0 0)"
    viewBox="0 0 9 12"
    width="14px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.46.001a.538.538 0 0 0-.358.174L.156 4.48a.538.538 0 1 0 .796.724l3.01-3.285v13.689a.563.563 0 0 0 .538.55.563.563 0 0 0 .538-.55V1.918l3.01 3.286a.538.538 0 1 0 .796-.724L4.898.175a.538.538 0 0 0-.437-.174z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`supports direction 1`] = `
<div>
  <svg
    class=""
    color="red"
    height="14px"
    transform="rotate(-90 0 0)"
    viewBox="0 0 9 12"
    width="14px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.46.001a.538.538 0 0 0-.358.174L.156 4.48a.538.538 0 1 0 .796.724l3.01-3.285v13.689a.563.563 0 0 0 .538.55.563.563 0 0 0 .538-.55V1.918l3.01 3.286a.538.538 0 1 0 .796-.724L4.898.175a.538.538 0 0 0-.437-.174z"
      fill="currentColor"
    />
  </svg>
</div>
`;
